packages/flutter/lib/src/foundation/consolidate_response.dart

```diff
@@ -6,19 +6,48 @@ import 'dart:async';
 import 'dart:io';
 import 'dart:typed_data';
 
-/// Efficiently converts the response body of an [HttpClientResponse] into a [Uint8List].
+/// Signature for getting notified when chunks of bytes are received while
+/// consolidating the bytes of an [HttpClientResponse] into a [Uint8List].
+///
+/// The `cumulative` parameter will contain the total number of bytes received
+/// thus far.
+///
+/// The `total` parameter will contain the _expected_ total number of bytes to
+/// be received (extracted from the value of the `Content-Length` HTTP response
+/// header), or -1 if the size of the response body is not known in advance.
+/// Even if the parameter reports a non-negative value, it is not 100%
```

Comment:
The `Content-Length` is the size of the transferred data, whether or not it is compressed, at least according to the [spec](https://tools.ietf.org/html/rfc2616#page-119). So this should be a good proxy for the progress, and I'm not sure that the comment is needed.  Do ask @Hixie, however, because he probably knows if the spec is implemented properly by most servers.

Reply:
I had added this comment to fix a bug previously.

Some servers in the wild won't set this properly.  And ones that do set it properly will give you the compressed size, but dart:io automagically decompresses for you without telling you whether it's compressed.  At one point I had patched this so that it would try to be smart and detect that, but that was abandoned since the header may not even be reliable.

Reply:
It may be worth documenting here that `total` could be less than `cumulative` when GZIP is used.

Do you think we could use the content-type header to determine if the response is one of our supported image formats or a likely compression  type?

Reply:
I updated the code to better handle gzipped content.  Now, `consolidateHttpClientResponseBytes()` will unzip the raw bytes itself if needed, which allowed us to turn off `autoUncompress` in the http client used by `NetworkImage`.

I also updated the docs.